### PR TITLE
Run on all devices even if other device types fail

### PR DESCRIPTION
**Pull request by ryanlai2**
This PR pulls out the code in EvaluateModels that loops through all inputBindingTypes  inputDataTypes and deviceCreationLocations into a separate function.

EvaluateModels loops through all deviceTypes and calls EvaluateModelGivenDeviceType

### CODE REVIEW

File: Tools/WinMLRunner/src/Run.cpp [hunk 1]
```diff
@@ -392,65 +448,21 @@ HRESULT EvaluateModels(const std::vector<std::wstring>& modelPaths, const std::v
                        << std::endl;
             continue;
         }
-
         for (const auto& deviceType : deviceTypes)
         {
-            for (const auto& inputBindingType : inputBindingTypes)
+            HRESULT evaluateModelGivenDeviceTypeResult =
+                EvaluateModelGivenDeviceType(model, deviceType, inputBindingTypes, inputDataTypes,
```
Comment (youngkim93) [inline Tools/WinMLRunner/src/Run.cpp]: Can we rename this to EvaluateModelWithDeviceType?

File: Tools/WinMLRunner/src/Run.cpp [hunk 2]
```diff
@@ -392,65 +448,21 @@ HRESULT EvaluateModels(const std::vector<std::wstring>& modelPaths, const std::v
...
-                                                                 deviceTypeStringified, inputDataTypeStringified,
-                                                                 inputBindingTypeStringified,
-                                                                 deviceCreationLocationStringified);
-                            }
-                        }
-                    }
-                }
+                lastEvaluateModelResult = evaluateModelGivenDeviceTypeResult;
+                std::cout << "Run failed for DeviceType: " << TypeHelper::Stringify(deviceType) << std::endl;
+                ;
```
Comment (youngkim93) [inline Tools/WinMLRunner/src/Run.cpp]: nit: extra ; :)

File: Tools/WinMLRunner/src/Run.cpp (hunk 1 above)
Comment (ryanlai2) [reply Tools/WinMLRunner/src/Run.cpp]: Sure sure! I can rename this.

File: Tools/WinMLRunner/src/Run.cpp (hunk 2 above)
Comment (ryanlai2) [reply Tools/WinMLRunner/src/Run.cpp]: Good catch!

**Comment by youngkim93** [comment]
It would be great if we can add test cases for this, but I understand that this might be hard to repro (since you should purposely fail in one of the devices, which is not what where should be at all)

**Comment by ryanlai2** [comment]
> It would be great if we can add test cases for this, but I understand that this might be hard to repro (since you should purposely fail in one of the devices, which is not what where should be at all)

Right, it would be great to add test cases for this. However, it would be hard to find a model that purposely fails on one of the devices but passes on others. 

In the future, I believe using mocking framework would be best to test this.

**Pull request merged**